options/options.go

```diff
@@ -614,7 +623,8 @@ func (interp *interpreter) interpretOptions(
 			return nil, err
 		}
 		if optn, ok := node.(*ast.OptionNode); ok {
-			if !uo.Name[0].GetIsExtension() && uo.Name[0].GetNamePart() == featuresFieldName {
+			optNodes = append(optNodes, optn)
```

Comment:
We accumulate all option nodes so that we can later find the best match if a required field in an option message was missing (below, line 658).